File: BE/Hinet.Service/Constant/NghiPhep/TrangThaiNghiPhepConstant.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Hinet.Service.Constant.NghiPhep
{
    public class TrangThaiNghiPhepConstant
    {
        [DisplayName("Tạo mới")]
        public static int TaoMoi => 0;
        [DisplayName("Đã gửi trưởng ban")]
        public static int DaGuiTruongBan => 1;
        [DisplayName("Trưởng ban phê duyệt")]
        public static int TruongBanDuyet => 2;
        [DisplayName("Gửi tổng giảm đốc")]
        public static int GuiTongGiamDoc => 3;
        [DisplayName("Tổng giám đốc phê duyệt")]
        public static int TongGiamDocPheDuyet => 4;
        [DisplayName("Trưởng ban từ chối")]
        public static int TruongBanTuChoi => 5;
        [DisplayName("Tổng giám đốc từ chối")]
        public static int TongGiamDocTuChoi => 6;
        [DisplayName("Hủy đơn đăng ký")]
        public static int HuyDangKy => 7;
    }
}

File: FE/src/constants/NghiPhep/TrangThaiNghiPhepConstant.ts
import { createConstant } from "../Constant";
const TrangThaiNghiPhepConstant = createConstant(
  {
    TaoMoi: 0,
    DaGuiTruongBan: 1,
    TruongBanDuyet: 2,
    GuiTongGiamDoc: 3,
    TongGiamDocPheDuyet: 4,
    TruongBanTuChoi: 5,
    TongGiamDocTuChoi: 6,
    HuyDangKy: 7,
  },
  {
    0: { displayName: "Tạo mới" },
    1: { displayName: "Đã gửi trưởng ban" },
    2: { displayName: "Trưởng ban phê duyệt" },
    3: { displayName: "Gửi tổng giám đốc" },
    4: { displayName: "Tổng giám đốc phê duyệt" },
    5: { displayName: "Trưởng ban từ chối" },
    6: { displayName: "Tổng giám đốc từ chối" },
    7: { displayName: "Hủy đăng ký" },
  }
);

export default TrangThaiNghiPhepConstant;
